Amazon.Ion.ObjectMapper.Test/IonObjectSerializerTest.cs

```diff
@@ -172,13 +172,14 @@ public void ExceptionOnMultiParameterIonPropertySetterMethods()
         public void SerializesAndDeserializesObjectsWithIonConstructor()
         {
             var serializer = new IonSerializer();
-            var wheel = new Wheel("default", 17, "MSW", "black");
+            var wheel = new Wheel("default", 17, 20, "MSW", "black");
```

Comment:
Should this just use Check?

Reply:
`Check` does not work here as the deserialized `Specification` property differs from the original (which is how we verify that the custom constructor was used).

I tried altering the Specification to be a field (which is ignored by default) and then let the custom constructor set the field but that does not work because the field is ignored during serialization and then absent entirely during deserialization (when the custom construction happens).

Reply:
Can you verify that they are not equal then, if that is expected?

Reply:
We check the deserialized specification more precisely with 
`Assert.AreEqual($"Specification: {wheel.Specification}, Size: {wheel.Size} inches", deserialized.Specification);` 
(which also entails checking that they are not equal).

The remaining checks for `offset`, `Brand`, and `color` are to verify that the respective method, property, and field logic in `DeserializeWithIonConstructor()` behaves as expected.